### PR TITLE
Heat Wave refactor

### PR DESCRIPTION
**Pull request by karandesai-96**
@hareshkh please update the description whenever possible :smile:

### CODE REVIEW

**Comment by hareshkh** [comment]
This PR mainly aims to set up the functionality of heat-wave weapon properly. Earlier only a single circle showed up in place of the heatwave which is now to be replaced by an array of 5 circles of different and changing radii and which have cuts in them for finger to pass through them.

File: core/src/com/sdsmdg/kd/gameplay/objects/HeatWave.java [hunk 1]
```diff
@@ -43,7 +47,7 @@ public void init(Magnus magnus) {
      * This method adds the velocity to radius of heatwave, hence expanding it.
      */
     public void expand(float delta) {
-        this.radius += this.velocity * delta;
+        this.radius += 0.3 * this.heatWaveNo * this.velocity * delta;
```
Comment (karandesai-96) [inline core/src/com/sdsmdg/kd/gameplay/objects/HeatWave.java]: This one for making inner waves fast right ? Nice way to incorporate it.

File: core/src/com/sdsmdg/kd/gameworld/GameRenderer.java [hunk 2]
```diff
@@ -76,13 +76,14 @@ public void render() {
...
-                for (int i = 0; i < 6; i++) {
+        for (int i = 0; i < gameWorld.nHeatWaves; i++) {
+            if (gameWorld.heatWaves[i].active) {
+                for (int j = 0; j < 11; j += 2) {
                     arcDrawer.begin(ArcDrawer.ShapeType.Line);
                     Gdx.gl.glLineWidth(64);
                     arcDrawer.setColor(new Color(0xffca28ff));
-                    arcDrawer.arc(heatwave.x, heatwave.y, heatwave.radius, 30 * (i + (i % 2)), 30, 50);
+                    arcDrawer.arc(gameWorld.heatWaves[i].x, gameWorld.heatWaves[i].y, gameWorld.heatWaves[i].radius,
+                            30 * (j + (i % 2)), 30, 50);
```
Comment (karandesai-96) [inline core/src/com/sdsmdg/kd/gameworld/GameRenderer.java]: Foreach loop looks cleaner here. You can use `heatwaveno` variable to mod it with 2.

File: core/src/com/sdsmdg/kd/gameworld/GameWorld.java [hunk 3]
```diff
@@ -173,7 +177,8 @@ public void update(float delta) {
     }
 
     public void selectWeapon() {
-        currentWeapon = 1 + random.nextInt(5);
+        //currentWeapon = 1 + random.nextInt(5);
+        currentWeapon = 2;
```
Comment (karandesai-96) [inline core/src/com/sdsmdg/kd/gameworld/GameWorld.java]: _TODO_: Revert it back before merging.

**Comment by hareshkh** [comment]
@karandesai-96, I think this PR has achieved its goal, and should be merged now. Please review the changes and merge.

File: core/src/com/sdsmdg/kd/gameplay/controllers/HeatWaveController.java [hunk 4]
```diff
@@ -1,27 +1,56 @@
...
 /**
  * @author Karan Desai
  */
 public class HeatWaveController {
-    private HeatWave heatWave;
+    private HeatWave[] heatWaves;
 
-    public HeatWaveController(HeatWave heatWave) {
-        this.heatWave = heatWave;
+    public HeatWaveController(HeatWave[] heatWave) {
```
Comment (karandesai-96) [inline core/src/com/sdsmdg/kd/gameplay/controllers/HeatWaveController.java]: Name this parameter `heatWaves`, for sanity.

**Comment by karandesai-96** [comment]
@hareshkh do this one change and merge it immediately. Everything looks good to merge.

**Pull request merged**